### ISSUE TITLE
Allow caching on cpu when compute some metrics.

### ISSUE THREAD

**Issue by kynk94** [PyTorchLightning/metrics#848]
## 🚀 Feature

Currently, some metrics like FID and KID only use GPU computing.
Since all datasets are cached in gpu memory, there is a big limitation on hardware.
It may be helpful to add `compute_on_cpu` option to compute these metrics like below.

```python
def update(self, imgs: Tensor, real: bool) -> None:  # type: ignore
    """Update the state with extracted features.

    Args:
        imgs: tensor with images feed to the feature extractor
        real: bool indicating if imgs belong to the real or the fake distribution
    """
    features = self.inception(imgs)

    if self.compute_on_cpu:
        features = features.detach().cpu()

    if real:
        self.real_features.append(features)
    else:
        self.fake_features.append(features)
```

### Motivation

I can compute FID on cloud system but not my local, because of the memory issue.
Currently, warn large memory foorprint. #468

**Comment by SkafteNicki** [comment]
Hi @kynk94,
I do not completely understand the problem here. If you initialize a metric, it will by default be on the CPU:
![image](https://user-images.githubusercontent.com/24896311/154930705-80c2ec53-452f-4349-adab-8ca0ee9de767.png)

Do you want to keep the overall metric on GPU, meaning that `features = self.inception(imgs)` is still calculated on the GPU but when you store the features for later you want that to be on CPU?

**Comment by Borda** [comment]
My understanding is that if you run training and you use all GPU memory it could be interesting to force/offload the metric computation to CPU... is it correct @kynk94?

**Comment by kynk94** [comment]
Yes, that's right.
If training on GPU, the inception features are stored in GPU memory to compute metric.
If have a lot of GPU, there are enough memory.
But most local desktops not have enough memory, and CPU memory capacity is larger than GPU memory.

In this case, to avoid memory overflow errors, need to cache the features on CPU memory.
